libyul/AsmAnalysis.cpp

```diff
@@ -244,6 +244,13 @@ bool AsmAnalyzer::operator()(VariableDeclaration const& _varDecl)
 {
 	bool success = true;
 	int const numVariables = _varDecl.variables.size();
+	if (m_resolver)
+		for (auto const& variable: _varDecl.variables)
+			m_resolver(
+				yul::Identifier{variable.location, variable.name},
```

Comment:
Not particularly happy about creating a local identifier here... we should probably define a proper class as interface for the resolver in the future instead of using a single lambda-function.

Reply:
Not too bad, but you should document why you are calling the resolver here :)